src/memory.c

```diff
@@ -109,3 +116,196 @@ void memory_free_page(void *pageaddr) {
     freemap[cellnumber] |= cellmask;
     pages_free++;
 }
+
+struct kmalloc_page_info * kmalloc_create_page_info(unsigned paddr) {
+    struct kmalloc_page_info *n = (struct kmalloc_page_info *)paddr;
+    n->max_free_gap = KMALLOC_NUM_SLOTS;
+    memset(n->free, 1, KMALLOC_SLOT_SIZE * KMALLOC_NUM_SLOTS);
+    return (struct kmalloc_page_info *)n;
+}
+
+void kmalloc_get_page() {
+    //get current process's pagetable
+    struct pagetable *pt = current->pagetable;
+    unsigned vaddr = kmalloc_next_vaddr;
```

Comment:
Good call. It was useful for seeing how the += PAGE_SIZE was affecting things during a debug print statement, but is unnecessary now. taken care of in upcoming commit.